Torantulino/Auto-GPT
# Added support for Google Custom Search API

### Description
This PR fixes https://github.com/Torantulino/Auto-GPT/issues/31

This PR adds functionality to the project to allow for a choice between the original Google search method and the Google Custom Search API. The google_search method uses the original method of scraping the HTML from the search results page, using googlesearch-python, while the google_official_search method uses the Google Custom Search API to retrieve search results. **Note** the Google Custom Search API provides 100 search queries per day for free. If you need more, you may sign up for [billing](https://cloud.google.com/billing/docs/how-to/manage-billing-account) in the API Console. Additional requests cost $5 per 1000 queries, up to 10k queries per day.

### How to test:

To test the functionality, ensure that you have valid API keys and search engine IDs for both the Google search method and the Google Custom Search API. You can set these values in your environment variables as described in the README.md file.

**Additional Notes:**

This pull request only adds functionality and makes improvements to existing code. No new features or major changes have been introduced.
